tls/extensions/s2n_server_certificate_status.c

```diff
@@ -13,16 +13,70 @@
  * permissions and limitations under the License.
  */
 
-#include "tls/s2n_cipher_suites.h"
 #include "tls/s2n_config.h"
 #include "tls/s2n_connection.h"
 #include "tls/s2n_tls.h"
 #include "tls/s2n_x509_validator.h"
 #include "tls/extensions/s2n_server_certificate_status.h"
 #include "utils/s2n_safety.h"
 
+#define U24_SIZE 3
+
+static int s2n_server_certificate_status_send_size(struct s2n_connection *conn)
+{
+    notnull_check(conn);
+    if (s2n_server_can_send_ocsp(conn)) {
+        return sizeof(uint8_t) + U24_SIZE + conn->handshake_params.our_chain_and_key->ocsp_status.size;
+    }
+
+    return 0;
+}
+
+int s2n_tls13_ocsp_extension_send_size(struct s2n_connection *conn)
+{
+    notnull_check(conn);
+    if (s2n_server_can_send_ocsp(conn)) {
+        uint16_t size = s2n_server_certificate_status_send_size(conn);
+        inclusive_range_check(0, size, 65535);
```

Comment:
Should this check be for `65535 - (2 * sizeof(uint16_t))`

Reply:
I think we can use GUARD_UINT16_THEN_INCREMENT here too. s2n_tls13_ocsp_extension_send_size() itself gets guarded when use